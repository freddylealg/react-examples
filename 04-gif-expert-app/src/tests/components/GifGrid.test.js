import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom';
import toJson from 'enzyme-to-json';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Prueba de GifGrid', () => {
    
    test('Probar snapshot ', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category="vegeta" />);
        expect( wrapper ).toMatchSnapshot(); 
    });


    test('Debe de mostar items cuando se cargan imÃ¡genes useGetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https:test.com',
            title: 'ejemplo'
        },
        {
            id: 'ABCD',
            url: 'https:test.com',
            title: 'ejemplo'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category="vegeta" />);

        // expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    })
    
    

});
