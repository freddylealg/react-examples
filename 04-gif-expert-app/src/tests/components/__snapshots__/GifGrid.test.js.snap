// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifGrid Probar snapshot  1`] = `
<Fragment>
  <h3>
    vegeta
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
